pkg/keys/keys.go

```diff
@@ -0,0 +1,79 @@
+package keys
+
+import (
+	"crypto"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/theupdateframework/go-tuf/data"
+)
+
+// SignerMap stores mapping between key type strings and signer constructors.
+var SignerMap sync.Map
+
+// Verifier stores mapping between key type strings and verifier constructors.
+var VerifierMap sync.Map
+
+var (
+	ErrInvalid    = errors.New("tuf: signature verification failed")
+	ErrInvalidKey = errors.New("invalid key")
+)
+
+// A Verifier verifies public key signatures.
+type Verifier interface {
+	// UnmarshalKey takes key data to a working verifier implementation for the key type.
+	// This performs any validation over the data.Key to ensure that the verifier is usable
+	// to verify signatures.
+	UnmarshalKey(key *data.Key) error
+
+	// MarshalKey returns the data.Key object associated with the verifier.
+	MarshalKey() *data.Key
+
+	// This is the public string used as a unique identifier for the verifier instance.
+	Public() string
+
+	// Verify takes a message and signature, all as byte slices,
+	// and determines whether the signature is valid for the given
+	// key and message.
+	Verify(msg, sig []byte) error
+}
+
+type Signer interface {
```

Comment:
add a convenience function to get `IDs()`?

Reply:
@trishankatdatadog @hosseinsia I haven't yet done this -- it's a tradeoff. You either call `signer.PublicData().IDs()`, or have every signer implementation implement the boiler plate code (since go won't have "base impl methods")

```
func (s someSigner) IDs() []byte {
  return s.PublicData().IDs()
}
```